CONTRIBUTING.md

```diff
@@ -19,25 +19,43 @@ reported the issue. Please try to include as much information as you can. Detail
 * Any modifications you've made relevant to the bug
 * Anything unusual about your environment or deployment
 
+To make it easier for you, we already provide an issue template when creating an 
+issue. It will be of great use to us if you can follow the template and capture as much detail as possible.
+
 
 ## Contributing via Pull Requests
 Contributions via pull requests are much appreciated. Before sending us a pull request, please ensure that:
 
-1. You are working against the latest source on the *master* branch.
+1. You are working against the latest source on the *development* branch.
 2. You check existing open, and recently merged, pull requests to make sure someone else hasn't addressed the problem already.
 3. You open an issue to discuss any significant work - we would hate for your time to be wasted.
 
+### Steps to contribute
 To send us a pull request, please:
 
 1. Fork the repository.
-2. Modify the source; please focus on the specific change you are contributing. If you also reformat all the code, it will be hard for us to focus on your change.
-3. Ensure local tests pass.
-4. Commit to your fork using clear commit messages.
-5. Send us a pull request, answering any default questions in the pull request interface.
-6. Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.
-
-GitHub provides additional document on [forking a repository](https://help.github.com/articles/fork-a-repo/) and 
-[creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
+2. Clone your forked repo
+````
+git clone git@github.com:<your-account>/ask-toolkit-for-vscode.git
+````
+3. Run `npm install` at the root of the repo, to install the extension 
+dependencies.
+4. Modify the source; please focus on the specific change you are contributing. If you also reformat all the code, it will be hard for us to focus on your change.
+5. Run the extension and check if the functionality is working as expected. You 
+can use the [`Extension` launch configuration](https://github.com/alexa/ask-toolkit-for-vscode/blob/development/.vscode/launch.json#L6) for running the extension host.
+6. Add test cases for your changes.
+7. Run the following extension tests and make sure they pass : 
+    - [`Extension Tests`](https://github.com/alexa/ask-toolkit-for-vscode/blob/development/.vscode/launch.json#L20) and [`Extension Tests (Coverage)`](https://github.com/alexa/ask-toolkit-for-vscode/blob/development/.vscode/launch.json#L39) launch configurations through the extension
+    - `npm test` using the terminal
```

Comment:
Running tests through command line should require developer to turn off all opened vscode instance, which I think not required here. It 's mainly servers for CI/CD purpose.  

I personally think Extension Test / Extension Test(coverage) is good enough for unit tests.

Reply:
- Let me add that info regarding open vscode instances. 
- As for which part is good enough, I would still leave it this way until we set the CI/CD kick the tests off in the extension.

Reply:
Added in the latest commit.